applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -0,0 +1,299 @@
+                                            const BoundedMatrix<double, TNumNodes, TNumNodes>& rCapacityMatrix,
+                                            double                                             DtTemperatureCoefficient)
+    {
+        rLeftHandSideMatrix = rConductivityMatrix;
+        rLeftHandSideMatrix += (DtTemperatureCoefficient * rCapacityMatrix);
+    }
+
+    void AddContributionsToRhsVector(VectorType&                                        rRightHandSideVector,
+                                     const BoundedMatrix<double, TNumNodes, TNumNodes>& rConductivityMatrix,
+                                     const BoundedMatrix<double, TNumNodes, TNumNodes>& rCapacityMatrix) const
+    {
+        const auto capacity_vector =
+                array_1d<double, TNumNodes>{-prod(rCapacityMatrix, GetNodalValuesOf(DT_TEMPERATURE))};
+        rRightHandSideVector = capacity_vector;
+        const auto conductivity_vector =
+                array_1d<double, TNumNodes>{-prod(rConductivityMatrix, GetNodalValuesOf(TEMPERATURE))};
+        rRightHandSideVector += conductivity_vector;
+    }
+
+    Vector CalculateIntegrationCoefficients(const Vector& rDetJContainer) const
+    {
+        const auto& r_integration_points = GetGeometry().IntegrationPoints(GetIntegrationMethod());
+
+        auto result = Vector{r_integration_points.size()};
+        for (unsigned int GPoint = 0; GPoint < r_integration_points.size(); ++GPoint) {
+            result[GPoint] = r_integration_points[GPoint].Weight() * rDetJContainer[GPoint];
+        }
+
+        return result;
+    }
+
+    BoundedMatrix<double, TNumNodes, TNumNodes> CalculateConductivityMatrix(const GeometryType::ShapeFunctionsGradientsType& rShapeFunctionGradients,
+                                                                            const Vector& rIntegrationCoefficients,
+                                                                            const ProcessInfo& rCurrentProcessInfo) const
+    {
+        GeoThermalDispersionLaw law{TDim};
+        const auto constitutive_matrix =
+                law.CalculateThermalDispersionMatrix(GetProperties(), rCurrentProcessInfo);
+
+        auto result = BoundedMatrix<double, TNumNodes, TNumNodes>{ZeroMatrix{TNumNodes, TNumNodes}};
+        for (unsigned int GPoint = 0; GPoint < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod()); ++GPoint) {
+            BoundedMatrix<double, TDim, TNumNodes> Temp = prod(constitutive_matrix, trans(rShapeFunctionGradients[GPoint]));
+            result += prod(rShapeFunctionGradients[GPoint], Temp) * rIntegrationCoefficients[GPoint];
+        }
+
+        return result;
+    }
+
+    BoundedMatrix<double, TNumNodes, TNumNodes> CalculateCapacityMatrix(const Vector& rIntegrationCoefficients) const
+    {
+        const auto& r_properties = GetProperties();
+        const auto  cWater = r_properties[POROSITY] * r_properties[SATURATION] *
```

Comment:
couple more inconsistencies in this function (c_water, c_solid, n_container, g_point, n)

Reply:
Done.